src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Block.cs

```diff
@@ -39,7 +39,9 @@ public override BoundNode VisitBlock(BoundBlock node)
                 }
             }
 
-            return new BoundBlock(node.Syntax, node.Locals.AddRange(additionalLocals), node.LocalFunctions, node.HasUnsafeModifier, instrumentation, builder.ToImmutableAndFree(), node.HasErrors);
+            var result = new BoundBlock(node.Syntax, node.Locals.AddRange(additionalLocals), node.LocalFunctions, node.HasUnsafeModifier, instrumentation, builder.ToImmutableAndFree(), node.HasErrors);
+            additionalLocals.Free();
+            return result;
```

Comment:
A using would have prevented this mistake. It feels like this compiler rule actively makes things harder and demonstrably (here and in other prs) causes more mistakes.